### PR TITLE
43 44 create user service and mongodb

### PR DESCRIPTION
**Pull request by akshaysharma21**
- Created MongoDB and user service
- Refactored directory structure to keep dockerfiles in a separate directory
- TODO: local testing using mongomock and monkeyfixtures
- to run in prod:
   navigate to the services directory
   build the docker containers using docker-compose build
   run the containers using docker-compose up
- Endpoints available for the user (currently):
   localhost/users: this will return a JSON string with a welcome message
   localhost/users/users (GET): this will return a list of mock users that I added to the database 
   localhost/users/users (POST): this will be used to add a new user to the database. 
   expects a JSON payload in the format { "name" : "<newName>",  "privateKey" : "<arbitrary key as a string>" }

** The PR is not ready to merge, it is a placeholder for feedback until I get the MongoDB fixtures right**

### CODE REVIEW

File: backend/services/users/src/routes.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        }
+        data.append(user)
+
+    return jsonify(
+        status=True,
+        data=data
+    )
+
+@app.route('/users', methods=['POST'])
+def createUser():
```
Comment (abhisachdev17) [inline backend/services/users/src/routes.py:35]: better endpoint name: /create (/users will already be in the path)

File: backend/services/users/src/routes.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        message="Hello Users of " + db.name
+    )
+
+
+@app.route("/users")
+def users():
+    print(db.values)
+    userList = db.users.find()
+    
+    user = {}
```
Comment (abhisachdev17) [inline backend/services/users/src/routes.py:19]: better endpoint name. /all maybe(?)

File: backend/services/users/src/routes.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    user = {
+        'name' : data['name'],
+        'privateKey' : data['privateKey']
+    }
+    db.users.insert_one(user)
+
+    return jsonify(
+        status = True,
+        message= 'User added successfully! :)'
+    ), 201
```
Comment (abhisachdev17) [inline backend/services/users/src/routes.py:46]: the http codes should be changed to use shared HTTP folder

**Comment by akshaysharma21** [comment]
Please note that the files that are in data directory are generated by mongodb for persistent storage in containers. It is currently a part of the pr so that everyone has some sample data to test and will be removed before the merge happens.

File: backend/services/users/src/__init__.py [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+app.config.from_object(config.Config(''))
+
+mongo = PyMongo(app)
+db = mongo.db
+
+userSchema = {
+    "$jsonSchema": {
+        "bsonType": "object",
+        "required": ["name", "privateKey"],
```
Comment (abhisachdev17) [inline backend/services/users/src/__init__.py:20]: private_key instead of privateKey

File: backend/services/users/src/routes.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+    print(db.values)
+    userList = db.users.find()
+    
+    user = {}
+    data = []
+
+    for usr in userList:
+        user = {
+            'name' : usr['name'],
+            'privateKey' : usr['privateKey']
```
Comment (ZackHolmberg) [inline backend/services/users/src/routes.py:25]: All of the key-related stuff, including all DB stuff, is gonna be scratched due to the direction shift. Sorry Akshay!

**Review by ZackHolmberg — CHANGES_REQUESTED** [review]

File: backend/services/users/src/routes.py (hunk 1 above)
Comment (akshaysharma21) [reply backend/services/users/src/routes.py:35]: Yeahh, th sames were placeholders, they def could use some better names lol

File: backend/services/users/src/routes.py (hunk 2 above)
Comment (akshaysharma21) [reply backend/services/users/src/routes.py:19]: Yesss

File: backend/services/users/src/routes.py (hunk 3 above)
Comment (akshaysharma21) [reply backend/services/users/src/routes.py:46]: Yeah, I was gonna change it once one of our prs get merged and I have ghat code to go

File: backend/services/users/src/__init__.py (hunk 4 above)
Comment (akshaysharma21) [reply backend/services/users/src/__init__.py:20]: Seems like we won't be needing this at all now

File: backend/services/users/src/routes.py (hunk 5 above)
Comment (akshaysharma21) [reply backend/services/users/src/routes.py:25]: Yeah, no problem! I had an early night yesterday and woke up to a bunch of messages, I'm trying to catch up but would you guys mind filling me in on the design direction during the next meeting as well?

**Pull request closed without merging**